tests/ops/op_math/test_controlled_decompositions.py

```diff
@@ -241,6 +241,47 @@ def test_trivial_ops_in_decomposition(self):
         assert len(decomp) == 5
         assert decomp == expected
 
+    @pytest.mark.parametrize(
+        "composite_op, want_decomp",
+        [
+            (
+                qml.ops.Prod(qml.PauliX(0), qml.PauliX(0)),  # type: ignore
+                [
+                    qml.CNOT(wires=[1, 0]),
+                    qml.CNOT(wires=[1, 0]),
+                ],
+            ),
+            (
+                qml.ops.SProd(1.23, qml.PauliX(0)),  # type: ignore
```

Comment:
I think I'd prefer to test something like `qml.s_prod(1j, qml.PauliX(0))` so that the operation remains unitary. While yes, this basically seems to work on the non-unitary operator, I'm not sure this is something we want to test or promise.

Reply:
Thanks, I think that's a good idea! 💡 (I also didn't know about `qml.s_prod()`.)